cms/migrations/0017_create_and_setup_home_page.py

```diff
@@ -0,0 +1,60 @@
+# Generated by Django 2.2.10 on 2020-06-09 10:12
+
+from django.db import migrations
+from django.contrib.contenttypes.models import ContentType
+from wagtail.core.models import Page, Site
+
+from cms.models import HomePage
+
+
+DEFAULT_HOMEPAGE_PROPS = dict(
+    title="MIT Bootcamps", tagline="Learn Innovation and Entrepreneurship from MIT"
+)
+DEFAULT_SITE_PROPS = dict(hostname="localhost", port=80)
+
+
+def create_home_page(apps, schema_editor):
+    """
+    Create the home page
+    """
+    site = Site.objects.filter(is_default_site=True).first()
+    valid_home_page = Page.objects.filter(
+        content_type=ContentType.objects.get_for_model(HomePage)
+    ).first()
+    root = Page.objects.get(depth=1)
+    if valid_home_page is None:
+        valid_home_page = HomePage(**DEFAULT_HOMEPAGE_PROPS)
+        root.add_child(instance=valid_home_page)
+        valid_home_page.refresh_from_db()
+    if site is None:
+        Site.objects.create(
+            is_default_site=True, root_page=valid_home_page, **DEFAULT_SITE_PROPS
+        )
+    elif site.root_page is None or site.root_page != valid_home_page:
+        site.root_page = valid_home_page
+        site.save()
+    wagtail_default_home_page = Page.objects.filter(
+        depth=2, content_type=ContentType.objects.get_for_model(Page)
+    ).first()
+    if wagtail_default_home_page is not None:
+        wagtail_default_home_page.delete()
```

Comment:
You need to transfer any existing child page under the new page before you delete the default home page otherwise this migration will empty out the CMS.

Reply:
Yes that can be done becuase that's true it make the cms empty and home page starts to give `404`.